Give more information on test failure.

kdelee opened the pull request:
Also allow user to disable test cleanup utilties. This makes any objects
created on the server and artifacts from scans persist. This can be useful when
looking into test failures or developing tests.

Closes #241

elyezer commented on CONTRIBUTING.rst:
```diff
@@ -19,3 +19,16 @@ was number 123. See the github documentation for more information on this
+disabled, run::
+
+    #Run a specific test with cleanup disabled
+    DEBUG=True py.test camayoc/tests/qpc/api/v1/authentication/test_login.py
```
I have some concerns about the debug flag:

* It will leave vCenter machines on and will require manual work to get them off and free up resource usage.
* It won't clear entities which may act as side effect for other tests. In addition to that on our automation job the slave is a throw away one so we won't be able to inspect the system there. If that is aiming local run than I would recommend using PDB.

elyezer reviewed (changes requested):
I don't think the DEBUG flag will be useful on automation and for local testing it can be replaced by using PDB and breakpoints. So I would recommend removing those changes and just focus on getting more information printed when a scan fail.

kdelee commented:
Sounds good that's not difficult to take out.

On Wed, Apr 18, 2018, 1:45 PM Elyézer Rezende <notifications@github.com>
wrote:

> *@elyezer* requested changes on this pull request.
>
> I don't think the DEBUG flag will be useful on automation and for local
> testing it can be replaced by using PDB and breakpoints. So I would
> recommend removing those changes and just focus on getting more information
> printed when a scan fail.
> ------------------------------
>
> In CONTRIBUTING.rst
> <https://github.com/quipucords/camayoc/pull/244#discussion_r182512456>:
>
> > @@ -19,3 +19,16 @@ was number 123. See the github documentation for more information on this
>  topic: https://help.github.com/articles/closing-issues-using-keywords/
>
>  If you need help figuring out how to do this, please comment on your PR! We can help or do it for you.
> +
> +When developing your tests, you can disable cleanup functions so any machines
> +that are turned on by scans stay on and all objects created on server persist.
> +This can be useful when running individual tests, but is not advisable for
> +running the whole test suite. To run a specific test with cleanup functionality
> +disabled, run::
> +
> +    #Run a specific test with cleanup disabled
> +    DEBUG=True py.test camayoc/tests/qpc/api/v1/authentication/test_login.py
>
> I have some concerns about the debug flag:
>
>    - It will leave vCenter machines on and will require manual work to
>    get them off and free up resource usage.
>    - It won't clear entities which may act as side effect for other
>    tests. In addition to that on our automation job the slave is a throw away
>    one so we won't be able to inspect the system there. If that is aiming
>    local run than I would recommend using PDB.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/quipucords/camayoc/pull/244#pullrequestreview-113322941>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Ab4n7zjHp-oJavxHjb0T0QwKcJM26bo5ks5tp3vAgaJpZM4TY6hW>
> .
>

kdelee commented:
@elyezer took out that extra code!

Now it is just the increased verbosity changes.

kdelee committed:
Give more information on test failure.

Have failed tests output more data that allows the user to find related
information in the logs based off of the scan id and scanjob id.

Closes #241

kdelee commented:
Looking at Jenkins this morning I realize I'd like to add some more verbosity to some other tests, so I pushed another commit.

kdelee committed:
Fix syntax error, variables are case sensitive.

Also fix copy-pasta problem -- after pausing a scan, we want to wait until it
is paused.

elyezer reviewed (approved):
LGTM
